modules/core/shared/src/main/scala-2/weaver/ExpectMacro.scala

```diff
@@ -39,18 +39,25 @@ private[weaver] object ExpectMacro {
   def allImpl(c: blackbox.Context)(values: c.Tree*): c.Tree = {
     import c.universe._
     val sourceLoc = new weaver.macros.Macros(c).fromContext.asInstanceOf[c.Tree]
-    val (cluesName, cluesValDef) = makeClues(c)
-    val clueMethodSymbol         = getClueMethodSymbol(c)
-
-    val transformedValues =
-      values.toList.map(replaceClueMethodCalls(c)(clueMethodSymbol,
-                                                  cluesName,
-                                                  _))
-    makeExpectations(c)(cluesName,
-                        cluesValDef,
-                        transformedValues,
-                        sourceLoc,
-                        q"None")
+    val total     = values.size
+    val allExpectations =
+      values.zipWithIndex.toList.map { case (value, index) =>
+        val (cluesName, cluesValDef) = makeClues(c)
+        val clueMethodSymbol         = getClueMethodSymbol(c)
```

Comment:
Do we need to instantiate those in each iteration?

Reply:
We need a unique `cluesName` and `cluesValDef`, but the `clueMethodSymbol` can be moved out.